### PR TITLE
TF RNN: Simplify slicing last output

### PR DESCRIPTION
**Pull request by farizrahman4u**

### CODE REVIEW

**Comment by farizrahman4u** [comment]
Oops, TF doesn't support -1 indices. Great.

**Comment by fchollet** [comment]
For sure it doesn't. You have to rely on symbolic shapes. Which implies building a shape tensor with `tf.pack`.

File: keras/backend/tensorflow_backend.py [hunk 1]
```diff
@@ -1202,10 +1202,16 @@ def _step(input, state):
...
+        ndim = len(outputs.get_shape()._dims)
+        slices = [slice(None)] * ndim
+        length = inputs.get_shape()[0]
+        if length:
+            slices[0] = length - 1
+            _outputs = outputs
+        else:
+            slices[0] = 0
+            _outputs = tf.reverse(_outputs, [True] + [False] * ndim)
+        last_output = _outputs[slices]
```
Comment (fchollet) [inline keras/backend/tensorflow_backend.py]: But now it's more code than the initial version, it's more difficult to follow, and I'm not convinced it's more efficient... (especially the `tf.reverse` seems like more way work)

Comment (farizrahman4u) [inline keras/backend/tensorflow_backend.py]: But it doesn't use reverse if time dimension is available. No. of lines are more, but ops are less.

Comment (farizrahman4u) [inline keras/backend/tensorflow_backend.py]: Hmm, but reverse must be much slower.. I don't think its worth it too..

**Comment by farizrahman4u** [comment]
Closing this for now.. Will open if I find something better.

**Pull request closed without merging**